packages/plugins/rows/src/editor/use-can-drop.ts

```diff
@@ -0,0 +1,46 @@
+import { useScopedStore } from '@edtr-io/core'
+import { findParent, getFocusPath, getFocusTree, Node } from '@edtr-io/store'
+import * as R from 'ramda'
+
+export function useCanDrop(id: string, draggingAbove: boolean) {
+  const store = useScopedStore()
+
+  return function(dragId?: string) {
+    return (
+      dragId &&
+      !wouldDropInOwnChildren(dragId) &&
+      !wouldDropAtInitialPosition(dragId)
+    )
+  }
+
+  function wouldDropInOwnChildren(dragId: string) {
+    const focusPath = getFocusPath(id)(store.getState()) || []
+    return focusPath.includes(dragId)
+  }
+
+  function wouldDropAtInitialPosition(dragId: string) {
+    const focusTree = getFocusTree()(store.getState())
+    if (!focusTree) return true
+    const parent = findParent(focusTree, dragId)
+
+    const dropIndex = getChildPosition(parent, id)
+    if (dropIndex === null) return true
```

Comment:
This needs to return false, because the check means the dropZone doesn't have the same parent as the dragged element. So it definitely is not in the same position. Maybe also add a comment, since that's not really clear by looking at the code.

Reply:
Done